lib/job-iteration/iteration.rb

```diff
@@ -206,11 +183,17 @@ def build_enumerator(params, cursor:)
     def assert_valid_cursor!(cursor)
       return if serializable?(cursor)
 
-      raise CursorError.new(
-        "Cursor must be composed of objects capable of built-in (de)serialization: " \
-        "Strings, Integers, Floats, Arrays, Hashes, true, false, or nil.",
-        cursor: cursor,
-      )
+      Deprecation.warn(<<~DEPRECATION_MESSAGE.chomp)
+        Cursor must be composed of objects capable of built-in (de)serialization:
+          Strings, Integers, Floats, Arrays, Hashes, true, false, or nil.
```

Comment:
Should we also allow to pass Symbol as part of the cursor?

Yesterday we noticed that having a Hash with keys as symbols also broke. Hash with symbols should be accepted as well 😄

Reply:
No. Symbols are not safe to serialize as JSON, because they are deserialized as Strings:

```sh
$ ruby -rjson -e 'p JSON.parse(JSON.dump(:symbol))'
"symbol"
```

`cursor_position` is serialized by the queue adapter (Resque/Sidekiq), not by ActiveJob itself, so none of Rails' magic symbol serialization happens.

Reply:
Thanks for the explanation